R/dotproduct.R

```diff
@@ -0,0 +1,73 @@
+#' @name dotproduct
+#' @title Calculate the dot product or normalized dot product
+#' @description Calculate the dot product (DP) or normalized dot product (NDP).
+#' `dotproduct` (with `normalize=TRUE`) returns a numeric 
+#'  value ranging between 0 and 1, where 0 
+#' indicates no similarity between the two MS/MS features, while 1 indicates 
+#' that the MS/MS features are identical. `dotproduct` (with `normalize=FALSE`)
+#' returns a numeric value with a lower bound of 0. 
+#' @usage dotproduct(x, y, m=0.5, n=2, normalize=TRUE)
+#' @param x `list`/`data.frame` of length 2 with m/z (`"mz"`) and corresponding 
+#' intensity values (`"intensity"`)
+#' @param y `list`/`data.frame` of length 2 with m/z (`"mz"`) and corresponding 
+#' intensity values (`"intensity"`)
+#' @param m `numeric(1)`, exponent for peak intensity-based weights
```

Comment:
I added the following to the details section: 

`m` and `n` are weights given on the peak intensity and the m/z values 
respectively. As default (`m = 0.5`), the square roots of the intensity 
values are taken to calculate weights. With increasing values for `m`, high
intensity values will be taken more into account for similarity calculation, 
i.e. differences between intensities will be intensified. 
With increasing values for `n`, high m/z values will be taken more into 
account for similarity calculation. Especially when working with small 
molecules, a value `n > 0` can be set, to give a weight on the m/z values to 
accommodate that shared fragments with higher m/z are less likely and will 
mean that molecules might be more similar. If `n != 0`, a warning will be 
raised if the corresponding m/z values are not identical, since small 
differences in m/z values will distort the similarity values with increasing
`n`. If `m=0` or `n=0`, intensity values or m/z values, respectively, are not 
taken into account.

@sgibb do you think this is sufficient?